### PR TITLE
docs: mention that vuex requires promise

### LINKED ISSUE #1002: Documentation: Missing reference for needed Promise polyfill
### What problem does this feature solve?
After starting using vuex (by the way: it's great!) I got a little bit checked when opening the IE 11 and getting  "[vuex] vuex requires a Promise polyfill in this browser."

I did read the installation guide in the documentation carefully, but it is not mentioned at all, that vuex depends on / requires a Promis ployfill, if it should work in IE. It would be great to include that in the documentation, along with tip, which polyfill to use and how to include it.

### What does the proposed API look like?
-

### PR DESCRIPTION
**Pull request by ktsn**
fix #1002

### CODE REVIEW

**Comment by akanshgulati** [comment]
@ktsn can you explain if its mandatory for Vuex to have Promises? Because I read it essential for actions only.

**Comment by ktsn** [comment]
@akanshgulati You cannot instantiate Vuex store since it has assertion in constructor.
https://github.com/vuejs/vuex/blob/dev/src/store.js#L19

**Comment by trainiac** [comment]
I tried using https://github.com/zloirock/core-js/blob/master/modules/es6.promise.js as a polyfill and found that it doesn't work. When promises are resolving I get that "Dont mutate vuex state outside of vuex". 

I dont want to suggest that vuex needs to support every promise polyfill but since this is what babel-env injects into your code by default I think it's worth noting that it is not compatible with vuex.

**Comment by ktsn** [comment]
I don't think it's caused by core-js...
If it was the case, it means core-js mutates the resolved value under the hood but it's doubtful. 
I tried it in jsfiddle but there are no errors. https://jsfiddle.net/m6nzkp4f/2/

**Comment by trainiac** [comment]
I'm not exactly sure how to cause it but they do mutate state under the hood. https://github.com/zloirock/core-js/blob/master/modules/es6.promise.js#L193.  But I agree your example isn't causing an error. I'll try to dig into my case more where I do get the error at that line number.

**Comment by trainiac** [comment]
My app is large and complicated. I've spent a few hours here trying to boil it down to what exactly the problem and can't seem to figure it out.  In then end, I don't believe that core-js promise and vuex are incompatible but there are ways that babel can be configured such that the promise library that it injects causes vuex mutations when using Promise.all 🤷‍♂️ . Ugh configuration hell.

File: docs/en/installation.md [hunk 1]
```diff
@@ -38,6 +38,31 @@ Vue.use(Vuex)
 
 You don't need to do this when using global script tags.
 
+### Promise
+
+Vuex requires [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises). If your supporting browsers do not implement Promise (e.g. IE), you can use a polyfill library such as [es6-promise](https://github.com/stefanpenner/es6-promise).
```
Comment (HerringtonDarkholme) [inline docs/en/installation.md]: Just a nit, comma before such as. https://www.grammarly.com/blog/such-as-comma/

Comment (ktsn) [reply docs/en/installation.md]: fixed.

**Pull request merged**